### ISSUE TITLE
Enable option --config on pulumi preview

### ISSUE THREAD

**Issue by naineel** [pulumi/pulumi#2975]
As of now pulumi up supports --config as an option but pulumi preview doesn't.
Can we enable this for pulumi preview as well ?

**Comment by lukehoban** [comment]
The current `pulumi up --config` flag results in doing two steps:
1. Modifying the local configuration for the active stack based on the args to `--config`
2. Then running the update

This is equivalent to `pulumi config set ...` then `pulumi up`.

The slight concern with `pulumi preview --config` is that it wouldn't really be a preview of the config changes - the `--config` part would have to actually modify the local configuration to match `pulumi up --config`.  We could certainly implement that, but I'm not entirely sure it's what's desired here.

More broadly - curious to understand your usecase for `pulumi up --config` in the first place.  This is generally uncommon, and was originally designed for use with `pulumi up <url> --config` for cases where you are deploying code that isn't in the local workspace.  For cases where you do have a local workspace, setting the configuration via `pulumi config set` or modifications to `Pulumi.<stackname>.yaml` should be the more directly supported options, and those compose well with preview/update as well.

**Comment by naineel** [comment]
Yes so the general idea is we want to separate config from code. I'm looking at options to do it.
So given I have the code available, at runtime I want to pass in a set of config variables to it.
So the config may not reside with the code at all. It will be pulled in from somewhere.
This is true for both `pulumi preview` and `pulumi up`. 
I expect to be able to use --config <key>=<value> multiple times to add config to the stack or override the default config I pull in at runtime with some dynamic logic.
Let me know if there is a way to do it currently and if this makes sense.

**Comment by geekflyer** [comment]
I'm achieving something similar via environment variables.

For example in our code we have stuff like this:

```
const assert = require('assert');

assert(process.env.DOCKER_IMAGE, 'required environment variable DOCKER_IMAGE did not get passed');

const dockerImage = process.env.DOCKER_IMAGE;

new k8s.v1.apps.Deployment('foo', {
...
image: dockerImage
});
```

then you run
```
DOCKER_IMAGE=foo-bar:v1.2.3 pulumi preview
```

**Comment by Antman261** [comment]
Surely there has to be an easier way to provide overrides at runtime than using an environment variable? I just want to specify an image tag when I run `pulumi up` ?

**Comment by lunaris** [comment]
I think we can close this as `preview` does support `--config` now (though there are challenges with that as tracked separately in e.g. #3776).
